media/system/js/share-uncompressed.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @package         Joomla.JavaScript
+ * @copyright       Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license         GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+/**
+ * Calls the sending process of the config class
+ */
+jQuery(document).ready(function ($)
+{
+    $('#sharebutton').click(function ()
+    {
+        var share_draft = {
+            articleId  : $().val(),
+            title: $('jform_title').val(),
+            sharetoken  : $().val(),
```

Comment:
Hmm you could add the sharetoken somewhere in a hidden field on the edit.php of com_content:

<input type="hidden" id="sharetoken" value="<?php echo $token; ?>" />

But as you generate it in your model, you should not need to have it here at all? You should return it after clicking / saving (json encoded) to built the link? 

Maybe i am missing something though.